test/pytest/test_browser.py

```diff
@@ -18,50 +20,97 @@
 }
 
 
-# Changed in version 3.10: Added the root_dir and dir_fd parameters.
-def glob_py310(fnmatch: str, *, root_dir: str, recursive: bool = False) -> Iterable[str]:
-    prefix = f'{root_dir}/'
-    prefixlen = len(prefix)
+@contextlib.asynccontextmanager
+async def spawn_test_server() -> AsyncIterator[URL]:  # noqa:RUF029
+    if 'COVERAGE_RCFILE' in os.environ:
+        coverage = ['coverage', 'run', '--parallel-mode', '--module']
+    else:
+        coverage = []
+
+    # pass the address through a separate fd, so that we can see g_debug() messages (which go to stdout)
+    addr_r, addr_w = os.pipe()
+    try:
+        server = subprocess.Popen(
+            [f'{BUILDDIR}/test-server', 'python3', '-m', *coverage, 'cockpit.bridge'],
+            env={**os.environ, 'TEST_SERVER_ADDRESS_FD': f'{addr_w}'},
+            stdin=subprocess.DEVNULL, pass_fds=(addr_w,), close_fds=True
+        )
+    except FileNotFoundError:
+        pytest.skip('No test-server')
+    os.close(addr_w)
+    address = os.read(addr_r, 1000).decode()
+    os.close(addr_r)
 
-    for result in glob.glob(f'{prefix}{fnmatch}', recursive=recursive):
-        assert result.startswith(prefix)
-        yield result[prefixlen:]
+    yield URL(address)
 
+    server.kill()
+    server.wait()
 
-@pytest.mark.parametrize('html', glob_py310('**/test-*.html', root_dir=f'{SRCDIR}/qunit', recursive=True))
-def test_browser(html):
-    if not os.path.exists(f'{BUILDDIR}/test-server'):
-        pytest.skip('no test-server')
+
+@contextlib.asynccontextmanager
+async def webdriver() -> AsyncIterator[WebdriverBidi]:
+    driver = ChromiumBidi(headless=os.environ.get('TEST_SHOW_BROWSER', '0') == '0')
```

Comment:
I never thought about running the unit tests with `T_S_B=1` (as the usual way to do that would be to start ./test-server and just open the html in firefox), but "because we can!" :grin:

Reply:
I actually feel like this should be part of the base WebdriverBidi class.  Possible follow-up.